### ISSUE TITLE
Cannot build a project template

### ISSUE THREAD

**Issue by jhanca-robotecai** [o3de/o3de#17802]
**Describe the bug**
Cannot build a project template downloaded from [o3de-extras](https://github.com/o3de/o3de-extras)

**Steps to reproduce**
Steps to reproduce the behavior (please find the [Dockerfile.txt](https://github.com/o3de/o3de/files/15066494/Dockerfile.txt)
attached - added extension to be able to upload it):
1. Pull o3de 
2. Pull o3de-extras
3. Register engine, gems
4. Create project from the template and try to build it
5. See the error

**Expected behavior**
It should be possible to build the project.

**Actual behavior**
`cmake` cannot find some files due to missing variables:
```
 CMake Error at /data/workspace/o3de/cmake/FileUtil.cmake:139 (file):
   file failed to open for reading (No such file or directory):
 
     /gem.json
 Call Stack (most recent call first):
   /data/workspace/o3de/cmake/FileUtil.cmake:153 (ly_file_read)
   /data/workspace/o3de/cmake/O3DEJson.cmake:28 (o3de_file_read_cache)
   /data/workspace/o3de/cmake/Gems.cmake:100 (o3de_read_json_array)
   /data/workspace/o3de/cmake/Gems.cmake:115 (get_all_gem_dependencies)
   /data/workspace/o3de/cmake/Gems.cmake:435 (get_all_gem_dependencies)
   /data/workspace/o3de/cmake/Subdirectories.cmake:392 (ly_enable_gems)
   /data/workspace/o3de/cmake/Subdirectories.cmake:448 (get_all_external_subdirectories_for_o3de_object)
   /data/workspace/o3de/cmake/Subdirectories.cmake:470 (get_external_subdirectories_in_use)
   /data/workspace/o3de/CMakeLists.txt:93 (add_subdirectory_on_external_subdirs) 
```

**Found in Branch**
`development`

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
o3de: c45de536f4
o3de-extras: c7b8a760

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Ubuntu 22.04

**Comment by nbbrooks** [comment]
I am encountering this as well, trying a workflow with the latest nightly binary and the development branch of o3de-extras (`c7b8a7`). Attempting to build a project using the manipulation template. Of note, there is a Gem/gem.json file relative to my $PROJECT_PATH where I am calling cmake from.

**Comment by jhanca-robotecai** [comment]
The issue is fixed by #17804 and might be closed after the latter is merged.

**Comment by michalpelka** [comment]
@nbbrooks Today, night build works fine with project templates.
